types/luxon/src/datetime.d.ts

```diff
@@ -1439,14 +1451,14 @@ export class DateTime {
      * @param opts - options that affect the creation of the Duration
      * @param opts.conversionAccuracy - the conversion system to use. Defaults to 'casual'.
      */
-    diffNow(unit?: DurationUnits, opts?: DiffOptions): Duration;
+    diffNow(unit?: DurationUnits, opts?: DiffOptions): Duration<Valid>;
 
     /**
      * Return an Interval spanning between this DateTime and another DateTime
      *
      * @param otherDateTime - the other end point of the Interval
      */
-    until(otherDateTime: DateTime): Interval;
+    until(otherDateTime: DateTime): IfValid<Interval<Valid>, DateTime<Invalid>, IsValid>;
```

Comment:
I think this should have been
```suggestion
    until(otherDateTime: DateTime): IfValid<Interval<Valid>, Interval<Invalid>, IsValid>;
```

Reply:
https://github.com/moment/luxon/blob/master/src%2Fdatetime.js#L2081

Probably an oversight in JS src, but I felt like I should be true to it.

Reply:
Oh wow. Reported as https://github.com/moment/luxon/issues/1551!